### PR TITLE
feat: Add log hints and add some missing log write events

### PR DESCRIPTION
**Pull request by JFreegman**
The hints will allow us to re-colourize the entire log in the future, and may also be useful for third-party tools

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/toxic/333)

### CODE REVIEW

File: src/prompt.c [hunk 1]
```diff
@@ -521,7 +521,7 @@ static void prompt_onConnectionChange(ToxWindow *self, Toxic *toxic, uint32_t fr
     if (connection_status != TOX_CONNECTION_NONE && Friends.list[friendnum].connection_status == TOX_CONNECTION_NONE) {
         msg = "has come online";
         line_info_add(self, c_config, true, nick, NULL, CONNECTION, 0, GREEN, "%s", msg);
-        write_to_log(ctx->log, c_config, msg, nick, true);
+        write_to_log(ctx->log, c_config, msg, nick, true, LOG_HINT_NAME);
```
Comment (robinlinden) [inline src/prompt.c:524]: Not `CONNECT`?

File: src/groupchats.c [hunk 2]
```diff
@@ -1711,12 +1721,18 @@ static void groupchat_onGroupNickChange(ToxWindow *self, Toxic *toxic, uint32_t
...
-    peer->name[length] = 0;
+    peer->name[length] = '\0';
     peer->name_length = length;
 
     line_info_add(self, toxic->c_config, true, peer->prev_name, peer->name, NAME_CHANGE, 0, MAGENTA,
                   " is now known as ");
 
+    ChatContext *ctx = self->chatwin;
+
+    char log_event[TOX_MAX_NAME_LENGTH * 2 + 32];
```
Comment (robinlinden) [inline src/groupchats.c:1732]: No need for `TOX_MAX_NAME_LENGTH * 2` when the previous name is added in write_to_log?

File: src/groupchats.c [hunk 3]
```diff
@@ -1672,14 +1676,20 @@ static void groupchat_onGroupSelfNickChange(ToxWindow *self, Toxic *toxic, uint3
...
 
     line_info_add(self, toxic->c_config, true, old_nick, chat->peer_list[peer_index].name, NAME_CHANGE, 0,
                   MAGENTA, " is now known as ");
 
     groupchat_update_last_seen(groupnumber, peer_id);
     groupchat_update_name_list(groupnumber);
+
+    ChatContext *ctx = self->chatwin;
+
+    char log_event[TOX_MAX_NAME_LENGTH * 2 + 32];
```
Comment (robinlinden) [inline src/groupchats.c:1690]: No need for `TOX_MAX_NAME_LENGTH * 2` when the previous name is added in write_to_log?

**Review by robinlinden — APPROVED** [review]

File: src/prompt.c (hunk 1 above)
Comment (JFreegman) [reply src/prompt.c]: Fixed

File: src/groupchats.c (hunk 2 above)
Comment (JFreegman) [reply src/groupchats.c]: True

**Pull request merged**